src/core/lib/gpr/sync_posix.cc

```diff
@@ -74,53 +72,32 @@ gpr_atm gpr_counter_atm_add = 0;
 #endif
 
 void gpr_mu_init(gpr_mu* mu) {
-#ifdef GRPC_ASAN_ENABLED
-  GPR_ASSERT(pthread_mutex_init(&mu->mutex, nullptr) == 0);
-  mu->leak_checker = (int*)gpr_malloc(sizeof(*mu->leak_checker));
-  GPR_ASSERT(mu->leak_checker != nullptr);
-#else
   GPR_ASSERT(pthread_mutex_init(mu, nullptr) == 0);
-#endif
+  LSAN_REGISTER_ROOT_REGION(mu, 1);
 }
 
 void gpr_mu_destroy(gpr_mu* mu) {
-#ifdef GRPC_ASAN_ENABLED
-  GPR_ASSERT(pthread_mutex_destroy(&mu->mutex) == 0);
-  gpr_free(mu->leak_checker);
-#else
   GPR_ASSERT(pthread_mutex_destroy(mu) == 0);
-#endif
+  //LSAN_UNREGISTER_ROOT_REGION(mu, 1);
```

Comment:
why is this commented? did you see test failures?

similarly this probably should be `REGION(mu, sizeof(*mu))`

Reply:
This is a mistake. I just wanted to do tests locally, but my environment doesn't support the `detect_leak`, and then forgot to restore it.

Reply:
Reverted.